crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs

```diff
@@ -78,18 +80,51 @@ impl Violation for YodaConditions {
     }
 }
 
-/// Return `true` if an [`Expr`] is a constant or a constant-like name.
-fn is_constant_like(expr: &Expr) -> bool {
+/// Comparisons left hand side must not be more [`ConstantLikelihood`] than their right hand side
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum ConstantLikelihood {
+    Unlikely = 0,
+    Probably = 1,   // CAMEL_CASED vars
+    Definitely = 2, // literals, empty dicts and tuples...
+}
+
+fn how_likely_constant_str(s: &str) -> ConstantLikelihood {
+    if str::is_cased_uppercase(s) {
+        ConstantLikelihood::Probably
+    } else {
+        ConstantLikelihood::Unlikely
+    }
+}
+
+/// Return [`Expr`] [`ConstantLikelihood`] level depending on simple heuristics.
+fn how_likely_constant(expr: &Expr) -> ConstantLikelihood {
+    if expr.is_literal_expr() {
+        return ConstantLikelihood::Definitely;
+    }
     match expr {
-        Expr::Attribute(ast::ExprAttribute { attr, .. }) => str::is_cased_uppercase(attr),
-        Expr::Tuple(ast::ExprTuple { elts, .. }) => elts.iter().all(is_constant_like),
-        Expr::Name(ast::ExprName { id, .. }) => str::is_cased_uppercase(id),
+        Expr::Attribute(ast::ExprAttribute { attr, .. }) => how_likely_constant_str(attr),
+        Expr::Name(ast::ExprName { id, .. }) => how_likely_constant_str(id),
+        Expr::Tuple(ast::ExprTuple { elts, .. }) | Expr::List(ast::ExprList { elts, .. }) => elts
```

Comment:
I think we should consider omitting lists and dictionaries, _or_ at the very least they should be added under preview mode to avoid new violations.

Reply:
Ahh, I guess the use of `list` was actually the motivating issue here. So, alternatively, can we try gating this whole change under `preview`, to avoid disruption for folks on stable?

Reply:
moved lists and dicts to preview